base/inference.jl

```diff
@@ -4743,23 +4749,30 @@ function inline_worthy(body::Array{Any,1}, src::CodeInfo, mod::Module,
     bodycost = 0
     for line = 1:length(body)
         stmt = body[line]
-        thiscost = statement_cost(stmt, src, mod, params)
+        if stmt isa Expr
+            thiscost = statement_cost(stmt, line, src, mod, params)::Int
+        elseif stmt isa GotoNode
```

Comment:
Any chance that it's better to do this by dispatch? I ask because `statement_cost` is recursive.